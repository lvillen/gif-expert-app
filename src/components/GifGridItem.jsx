import React from 'react';
import PropTypes from 'prop-types';
import 'animate.css';
import { AddCategory } from './AddCategory';

export const GifGridItem = ({ title, url }) => {
  return (
    <>
      <div className="card animate__animated animate__fadeInUp">
        <img src={ url } alt={ title } />
        <p> { title } </p>
      </div>
    </>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}