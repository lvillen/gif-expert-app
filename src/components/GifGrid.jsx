import { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoryProp }) => {
  const [images, setImages] = useState([]);

  useEffect( () => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=Lord+of+the+Rings&limit=10&api_key=wOYIa1qIXGN2erRRDRcJ8KWrqPA6XiyI';
    const response = await fetch(url);
    const { data } = await response.json();

    const gifs = data.map( img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      };
    });

    setImages( gifs );
  }

  return (
    <>
      {console.log(images)}
      <h3>{ categoryProp }</h3>
      {
        images.map( img => (
          <GifGridItem 
            key={img.id} 
            { ...img } 
          />
        ))
      }
    </>
  );
};
