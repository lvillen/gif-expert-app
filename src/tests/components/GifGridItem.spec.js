import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests GifGridItem Component', () => {
  const title = 'A title';
  const url = 'https://localhost/image.jpg';
  const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );

  it('should match the snapshot', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  it('should have a p tag with title', () => {
    const p = wrapper.find('p');
    
    expect( p.text().trim() ).toBe( title );
  });

  it('should have an img tag that matches url and must have an alt attribute with the title', () => {
    const img = wrapper.find('img');

    expect( img.prop('src') ).toBe( url );
    expect( img.prop('alt') ).toBe( title );
  });

  it('should have animate__fadeIn in div class', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');

    expect( className.includes('animate__fadeIn') ).toBe( true );
  })
});