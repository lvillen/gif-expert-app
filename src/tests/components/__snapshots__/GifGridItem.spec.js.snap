// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem Component should match the snapshot 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeInUp"
  >
    <img
      alt="A title"
      src="https://localhost/image.jpg"
    />
    <p>
       
      A title
       
    </p>
  </div>
</Fragment>
`;
